crates/ruff_macros/src/map_codes.rs

```diff
@@ -113,9 +113,23 @@ pub(crate) fn map_codes(func: &ItemFn) -> syn::Result<TokenStream> {
                     Self::#linter(linter)
                 }
             }
+
+            // Rust doesn't yet support `impl const From<RuleCodePrefix> for RuleSelector`
+            // See https://github.com/rust-lang/rust/issues/67792
             impl From<#linter> for crate::rule_selector::RuleSelector {
                 fn from(linter: #linter) -> Self {
-                    Self::Prefix{prefix: RuleCodePrefix::#linter(linter), redirected_from: None}
+                    let prefix = RuleCodePrefix::#linter(linter);
+                    if is_single_rule_selector(&prefix) {
+                        Self::Rule {
+                            prefix,
+                            redirected_from: None,
+                        }
+                    } else {
+                        Self::Prefix {
+                            prefix,
+                            redirected_from: None,
+                        }
+                    }
                 }
```

Comment:
We could move all of this logic out of the macro if we instead implement `impl From<#linter> for RuleCodePrefix` because the `RuleCodePrefix::#linter(linter)` code is the only code that requires code-gen

Reply:
Interesting. I'd like to separate further changes to the macro from this pull request though, if that's okay?

Reply:
Yeah sure. I should have written this more clearly. This is feedback based on last week's discussion on how it probably would be possible to move a lot out of the macros if we abstracted the dynamic parts by traits instead of where we generate the implementations for our types. But this isn't something that concerns this PR. I just noticed it as a good example because the dynamic part is limited to a single line.